### PR TITLE
Persist stored cards Redux state

### PR DESCRIPTION
**Pull request by lamosty**
Part of #7329. Added a schema for stored cards items so we can persist them.

/cc @gwwar 

Test live: https://calypso.live/?branch=add/persist-stored-cards

### CODE REVIEW

**Comment by gwwar** [comment]
Nice work @lamosty! This works really well. Do you mind adding a few persistence tests? See [this](https://github.com/Automattic/wp-calypso/blob/master/client/state/products-list/test/reducer.js#L68) for an example

File: client/state/stored-cards/reducer.js [hunk 1]
```diff
@@ -26,25 +28,14 @@ import {
...
  * @return {Array}         Updated state
  */
-export const items = ( state = [], action ) => {
-	switch ( action.type ) {
-		case STORED_CARDS_ADD_COMPLETED:
-			return [ ...state, action.item ];
-
-		case STORED_CARDS_FETCH_COMPLETED:
-			return action.list;
+export const items = createReducer( [], {
```
Comment (gziolo) [inline client/state/stored-cards/reducer.js]: 👍

File: client/state/stored-cards/reducer.js [hunk 2]
```diff
@@ -26,25 +28,14 @@ import {
...
+	[ STORED_CARDS_ADD_COMPLETED ]: ( state, action ) => [ ...state, action.item ],
 
-		case STORED_CARDS_DELETE_COMPLETED:
-			return state.filter( item => item.stored_details_id !== action.card.stored_details_id );
+	[ STORED_CARDS_FETCH_COMPLETED ]: ( state, action ) => action.list,
```
Comment (gziolo) [inline client/state/stored-cards/reducer.js]: Those action handlers look very good. You can also consider using ES6 destructuring like:
```js
( state, { item } ) => [ ...state, item ]
( state, { list } ) => list
```
Totally up to you :)

Comment (lamosty) [inline client/state/stored-cards/reducer.js]: Great idea, thanks!

**Comment by lamosty** [comment]
Thanks @gwwar! Added the persistence tests and also updated the schema a bit to reflect the fixtures. Could you give it another review, please?

**Comment by gziolo** [comment]
LGTM 👍

**Comment by gwwar** [comment]
LGTM too! :shipit:

**Comment by lamosty** [comment]
Thanks for the reviews @gziolo and @gwwar!

**Pull request merged**